services/job_subscriber.go

```diff
@@ -2,24 +2,47 @@ package services
 
 import (
 	"sync"
+	"time"
 
 	"github.com/smartcontractkit/chainlink/logger"
 	"github.com/smartcontractkit/chainlink/store"
 	"github.com/smartcontractkit/chainlink/store/models"
+	"github.com/smartcontractkit/chainlink/utils"
 	"go.uber.org/multierr"
 )
 
 // JobSubscriber listens for push notifications from the ethereum node's
 // websocket for specific jobs.
-type JobSubscriber struct {
+type JobSubscriber interface {
+	AddJob(job models.JobSpec, bn *models.IndexableBlockNumber) error
+	Connect(bn *models.IndexableBlockNumber) error
+	Disconnect()
```

Comment:
`type HeadTrackable interface` already has `Connect`, `Disconnect`, and `OnNewHead`. You can collapse this with:
```
type JobSubscriber interface {
  HeadTrackable
  AddJob(job models.JobSpec, bn *models.IndexableBlockNumber) error
  ...
}
```

Reply:
We could still DRY this up by using `HeadTrackable`

Reply:
Right, I'd tried HeadTracker, but forgot about HeadTrackable. Will do.